### PR TITLE
Small API changes to allow client applications to use google APIs

### PR DESCRIPTION
**Pull request by urubatan**
This pull request is backwards compatible, but has also some small API changes to allow client applications to use google APIs after the user authenticate.
The client applications have the options to add extra scopes to the authentication, and also to request a refresh_token to work "offline"

### CODE REVIEW

**Review by jeremy — COMMENTED** [review]
General-purpose OAuth2 authorization is out of scope for this project. This is targeted at Google Sign-In support only.

File: app/controllers/google_sign_in/authorizations_controller.rb [hunk 1]
```diff
@@ -4,13 +4,17 @@ class GoogleSignIn::AuthorizationsController < GoogleSignIn::BaseController
   skip_forgery_protection only: :create
 
   def create
-    redirect_to login_url(scope: 'openid profile email', state: state),
+    redirect_to login_url(scope: "openid profile email #{GoogleSignIn.extra_scopes&.join(" ")}", state: state),
```
[the remark this answers is not in the document]
Comment (urubatan) [reply app/controllers/google_sign_in/authorizations_controller.rb:7]: that is a string not an array of strings

File: app/controllers/google_sign_in/callbacks_controller.rb [hunk 2]
```diff
@@ -3,35 +3,51 @@
...
+    token.token
+  end
+
+  def refresh_token
+    token.refresh_token
+  end
+
+  def error_message_for(error_code)
+    error_code.presence_in(GoogleSignIn::OAUTH2_ERRORS) || 'invalid_request'
+  end
```
[the remark this answers is not in the document]
Comment (urubatan) [reply app/controllers/google_sign_in/callbacks_controller.rb:52]: sorry, I'll fix my editor settings

**Comment by urubatan** [comment]
thanks for the comments, since it is out of scope please ignore

**Pull request closed without merging**